lib/http.c

```diff
@@ -3734,6 +3734,8 @@ CURLcode Curl_http_readwrite_headers(struct Curl_easy *data,
           }
         }
 
+        k->httpcode_1xx = k->httpcode >= 100 && k->httpcode < 200;
```

Comment:
How about instead setting the bit if it was a 1xx, 204 or 304 response? And probably also if it is a `NOBODY` request (which sends a `HEAD` and thus also doesn't have a body).